tst/EnergyPlus/unit/HeatBalFiniteDiffManager.unit.cc

```diff
@@ -233,4 +237,106 @@ TEST_F(EnergyPlusFixture, HeatBalFiniteDiffManager_adjustPropertiesForPhaseChang
+     "   0.9, !- Thermal Absorptance",
+     "   0.85, !- Solar Absorptance",
+     "   0.85;                    !- Visible Absorptance",
+     "Material:AirGap,",
+     "   F05 Ceiling air space resistance, !- Name",
+     "   0.18;                    !- Thermal Resistance{ m2 - K / W }",
+     "Material:NoMass,",
+     "   CP02 CARPET PAD, !- Name",
+     "   Smooth, !- Roughness",
+     "   0.1, !- Thermal Resistance{ m2 - K / W }",
+     "   0.9, !- Thermal Absorptance",
+     "   0.8, !- Solar Absorptance",
+     "   0.8;                     !- Visible Absorptance",
+        
+     "Material,",
+     "   F16 Acoustic tile, !- Name",
+     "   MediumSmooth, !- Roughness",
+     "   0.0191, !- Thickness{ m }",
+     "   0.06, !- Conductivity{ W / m - K }",
+     "   368, !- Density{ kg / m3 }",
+     "   590.000000000002, !- Specific Heat{ J / kg - K }",
+     "   0.9, !- Thermal Absorptance",
+     "   0.3, !- Solar Absorptance",
+     "   0.3;                     !- Visible Absorptance",
+        
+     "Material,",
+     "   M11 100mm lightweight concrete, !- Name",
+     "   MediumRough, !- Roughness",
+     "   0.1016, !- Thickness{ m }",
+     "   0.53, !- Conductivity{ W / m - K }",
+     "   1280, !- Density{ kg / m3 }",
+     "   840.000000000002, !- Specific Heat{ J / kg - K }",
+     "   0.9, !- Thermal Absorptance",
+     "   0.5, !- Solar Absorptance",
+     "   0.5;                     !- Visible Absorptance",
+     
+     "Construction,",
+     "  ExtSlabCarpet 4in ClimateZone 1 - 8, !- Name",
+     "  MAT - CC05 4 HW CONCRETE, !- Outside Layer",
+     "  CP02 CARPET PAD;         !- Layer 2",
+     "Construction,",
+     "   Interior Floor, !- Name",
+     "   F16 Acoustic tile, !- Outside Layer",
+     "   F05 Ceiling air space resistance, !- Layer 2",
+     "   M11 100mm lightweight concrete;  !- Layer 3",
+     "Construction:AirBoundary,",
+     "   Air Wall_ConstructionAirBoundary,  !- Name",
+     "   None,                    !- Air Exchange Method",
+     "   0;                       !- Simple Mixing Air Changes per Hour {1 / hr}",
+     "Output:Constructions,",
+     "Constructions;",
+     "Output:Constructions,",
+     "Materials;", });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+          
+    ErrorsFound = false;
+    GetMaterialData(*state, ErrorsFound); // read material data
+    EXPECT_FALSE(ErrorsFound);    // expect no errors
+
+    ErrorsFound = false;
+    GetConstructData(*state, ErrorsFound); // read construction data
+    EXPECT_FALSE(ErrorsFound);     // expect no errors
+
+    // allocate properties for construction objects when it is used or not for building surfaces in the model
+    
+    state->dataConstruction->Construct(1).IsUsed=false;
+    state->dataConstruction->Construct(2).IsUsed = true;
+    state->dataConstruction->Construct(3).IsUsed = true;
+
+    //call the function for initialization of finite difference calculation
+    InitialInitHeatBalFiniteDiff(*state);  
```

Comment:
Much better!  OK, I'll build locally but this is now on track to go in.  Thanks for the quick fix up.